added startsWith and endsWith

martinflory opened the pull request:
Ok...
- rolled back to .js, and used the underscore-string functions for `startsWith` and `endsWith`.
- made the linter work with .js (for this project for now).
- added some tests for this functionality.

Our linter module is private, we should probably remove it from here, or make it public? If we think using it for Baseline (either .js or .es) is a good idea, then maybe make public. What do you think @pepoviola @dgbeck ?

Commit:
:hammer: added startsWith and endsWith

Commit:
:hammer: removed position in the helper functions

Commit:
:hammer: removed lib es6 version

Commit:
cleanup test

pepoviola commented on package.json:
```diff
@@ -2,9 +2,14 @@
   "version": "1.1.5",
   "main": "matchesWhereQuery.js",
+  "scripts": {
+    "test": "mocha tests/test.js --reporter spec"
```
We should setup the CI to run this test, and also we can add more test to cover all cases.
Thx!

martinflory replied:
Yes, thanks! Its working fine now and with the tests. :)

pepoviola commented on matchesWhereQuery.js:
```diff
@@ -3,45 +3,73 @@ var _ = require( 'underscore' );
-		var thisKey = whereQueryKeys[ i ];
-		var thisKeyQuery = whereQuery[ thisKey ];
-		var thisObjectValue = object[ thisKey ];
+		var attribute = whereQueryKeys[ i ];
```
I think we can use another name for this `var`. I personally prefer `key` over `attribute`. What did you think @dgbeck .
Thx!

dgbeck replied:
seems pretty similar to me, no preference, my head spins for all of these names!

pepoviola replied:
Oks, sounds good. 👍

martinflory replied:
Talked to Javi and changed it to attributeName. :)

pepoviola reviewed (commented):
Looks good @martinflory, small comments inline.
Thx!

martinflory commented:
Oh, I just found a case in which it tries to apply the rules for es6 in files to be parsed as es5. :( So eslint might not be as easy as I thought.

Commit:
removing eslint for js, needs more work

dgbeck commented on matchesWhereQuery.js:
```diff
@@ -3,45 +3,73 @@ var _ = require( 'underscore' );
 };
+
+function _startsWith( str, starts ) {
+	str = _makeString( str );
```
we already know this is a string, because we check it on li nne 38

martinflory replied:
So true, thanks! Fixed!

dgbeck commented on matchesWhereQuery.js:
```diff
@@ -3,45 +3,73 @@ var _ = require( 'underscore' );
+}
+
+function _makeString( object ) {
+	if( ! object ) return '';
```
same as above, we know this is a string

martinflory replied:
same! fixed, thx!

dgbeck commented on matchesWhereQuery.js:
```diff
@@ -3,45 +3,73 @@ var _ = require( 'underscore' );
+				break;
+			case 'startsWith':
+				// eslint-disable-next-line max-len
+				if( ! _.isString( objectAttribute ) || ! _.isString( queryAttribute.value ) ) throw new Error( 'For startsWith comparator both the object attribute and the query value must be strings.' );
```
i think we might want to consider the `null` case here (as well as with `endsWith`). if the `objectAttribute` is `null` that should be a non-match

martinflory replied:
Good one, its done, and tested! thanks

dgbeck commented:
hi @martinflory comments inline, thx!

pepoviola committed:
Circleci project setup (#7)

* Add .circleci/config.yml

* install only prod

Commit:
:ok_hand: fixes based on PR

- Removed eslint module and dep
- Removed `--production` label from npm install in the circleCI config
- Added validation for null objectAttribute for string comparisons

Commit:
:hammer: testing, testing, testing

pepoviola commented:
Awesome, I like the test!! ping @dgbeck  to merge and publish.
Thx!

dgbeck commented:
awesome, thx!

dgbeck commented on matchesWhereQuery.js:
```diff
@@ -3,45 +3,69 @@ var _ = require( 'underscore' );
 };
+
+function _startsWith( str, starts ) {
+	starts = String( starts );
```
@martinflory we are already checking that this is a string... why cast it here?

martinflory replied:
Yes! You're right. thanks

dgbeck commented on matchesWhereQuery.js:
```diff
@@ -3,45 +3,69 @@ var _ = require( 'underscore' );
+}
+
+function _endsWith( str, ends ) {
+	ends = String( ends );
```
same here

martinflory replied:
Yes! You're right. thanks

dgbeck commented:
@martinflory one comment then good to merge, thx!

Commit:
:hammer: removing unneeded string check

martinflory commented:
All sorted! thanks for you corrections and suggestions @dgbeck @pepoviola! Merging now.
